admin/app/components/solidus_admin/orders/cart/component.html.erb

```diff
@@ -0,0 +1,115 @@
+<div
+  class="w-full relative overflow-visible"
+  data-controller="<%= stimulus_id %>"
+  data-<%= stimulus_id %>-products-url-value="<%= solidus_admin.variants_for_order_path(@order) %>"
+  data-action="
+    keydown-><%= stimulus_id %>#selectResult
+  "
+>
+  <%= render component('ui/panel').new do |panel| %>
+    <div>
+      <div class="peer">
+      <%= render component("ui/forms/search_field").new(
+        "data-action": "#{stimulus_id}#search focus->#{stimulus_id}#showResults #{stimulus_id}#showResults",
+        placeholder: t(".search_placeholder"),
+        "data-#{stimulus_id}-target": "searchField",
+      ) %>
+      </div>
+
+      <%# results popover %>
+      <details class="px-6 relative overflow-visible">
+        <summary class="hidden"></summary>
+        <div
```

Comment:
A view component would be great.

Reply:
As above, we're driving the component generation/extraction from bottom up, "generalizing" too early means missing what the "right" parameters are, that's a risk also with when you have 2-3 concrete cases to extract from but is much closer to something truly reusable.

The _section comments_ like `# result popover` are a clear hint that you correctly noticed, but we need to hold a little bit more 😊

Reply:
This makes total sense.